typing/env.ml

```diff
@@ -282,27 +284,19 @@ module TycompTbl =
              | [] -> f name None
              | (_, hidden, _) :: _ -> f name (Some (desc, hidden)))
 
-    let rec find_all ~mark name tbl =
-      List.map (fun (id, desc) -> Pident id, desc, nothing)
-        (Ident.find_all name tbl.current) @
-      match tbl.layer with
-      | Nothing -> []
-      | Lock {next; _} -> find_all ~mark name next
-      | Open {using; next; components; root} ->
-          let rest = find_all ~mark name next in
-          let using = if mark then using else None in
-          match NameMap.find name components with
-          | exception Not_found -> rest
-          | opened ->
-              List.map
-                (fun desc -> Pdot (root, name), desc,
-                  mk_callback rest name desc using)
-                opened
-              @ rest
-
     let add_lock lock next =
       { current = Ident.empty; layer = Lock {lock; next} }
 
+    let add_if_staged_lock lock next =
```

Comment:
Probably better to move this function outside of the module, since this `module Idtbl` is polymorphic over locks.

(But it will probably just be removed, per suggestions below)